server/src/main/java/org/opensearch/repositories/RepositoriesService.java

```diff
@@ -261,6 +273,29 @@ public boolean mustAck(DiscoveryNode discoveryNode) {
         );
     }
 
+    private void validateRestrictedSystemRepositorySettingsAndType(
+        RepositoryMetadata currentRepositoryMetadata,
+        RepositoryMetadata newRepositoryMetadata
+    ) {
+        if (SYSTEM_REPOSITORY_SETTING.get(currentRepositoryMetadata.settings())) {
+            if (newRepositoryMetadata.type() != currentRepositoryMetadata.type()) {
+                throw new RepositoryException(
+                    currentRepositoryMetadata.name(),
+                    "trying to modify system repository type " + currentRepositoryMetadata.type()
+                );
+            }
+            Repository repository = repositories.get(currentRepositoryMetadata.name());
+            for (Setting<?> setting : repository.restrictedSystemRepositorySettings()) {
```

Comment:
`REMOTE_STORE_INDEX_SHALLOW_COPY` setting will be in snapshot repository. for that, we need to add validation in non system repository as well. or should we make snapshot repository as well a system repository once user enables shallow snapshots on it?